src/MyInfoGovClient.class.ts

```diff
@@ -1,609 +1,400 @@
+      redirect_uri: redirectEndpoint ?? this.redirectEndpoint,
+      sp_esvcId: singpassEserviceId ?? this.singpassEserviceId,
     })
+    return `${this.baseAPIUrl}${Endpoint.Authorise}?${queryParams}`
+  }
 
-    // Generate the request signature by signing the constructed basestring
-    // with private key
-    const signature = this._signBaseString(
-      basestring,
-      this.privateKey,
-      'base64',
+  /**
+   * Retrieves the given MyInfo attributes from the Person endpoint after
+   * the client has logged in to SingPass and consented to providing the given
+   * attributes.
+   * @param accessToken Access token given by MyInfo
+   * @param requestedAttributes Attributes to request from Myinfo. Should correspond
+   * to the attributes provided when initiating SingPass login.
+   * @param singpassEserviceId Optional alternative e-service ID.
+   * Defaults to the e-serviceId provided in the constructor.
+   * @returns Object containing the user's NRIC/FIN and the data
+   * @throws {InvalidTokenSignatureError} Throws if the JWT signature is invalid
+   * @throws {WrongAccessTokenShapeError} Throws if decoded JWT has an unexpected
+   * type or shape
+   * @throws {MyInfoResponseError} Throws if MyInfo returns a non-200 response
+   * @throws {DecryptDataError} Throws if an error occurs while decrypting data
+   * @throws {InvalidDataSignatureError} Throws if signature on data is invalid
+   * @throws {WrongDataShapeError} Throws if decrypted data from MyInfo is
+   * of the wrong type
+   */
+  async getPerson(
+    accessToken: string,
+    requestedAttributes: MyInfoScope[],
+    singpassEserviceId?: string,
+  ): Promise<IPersonResponse> {
+    // Extract NRIC
+    const uinFin = this.extractUinFin(accessToken)
+    // Get Person data
+    const data = await this._sendPersonRequest(
+      accessToken,
+      requestedAttributes,
+      singpassEserviceId ?? this.singpassEserviceId,
+      uinFin,
     )
+    return { uinFin, data }
+  }
 
-    // Construct the authentication header using the signature, nonce and
-    // timestamp
-    const authHeader = this._formulateAuthHeader({
-      realm: this.realm,
-      appId: this.appId,
-      nonce,
-      signature,
-      timestamp,
-    })
-
-    // Construct the request headers
-    const headers = {
-      'Content-Type': 'application/json',
-      Authorization: authHeader,
-      Accept: 'application/json',
-    }
-
-    // Construct the querystring params
-    const querystring = {
-      attributes: requestedAttributes.join(),
+  /**
+   * Requests MyInfo attribute data from the Person endpoint.
+   * @param accessToken Access token provided by Token endpoint
+   * @param requestedAttributes Attributes to request from MyInfo, which
+   * user has consented to provide
+   * @param singpassEserviceId ID registered with SingPass
+   * @param uinFin Optional uinFin if it has already been decoded. If not
+   * given, it is extracted from the access token.
```

Comment:
Can I check what's the use case for overriding the UIN/FIN? Would it be simpler to remove this functionality?

Reply:
yeah you're right, this is a bit of an accident of history. let me merge `getPerson` and `_sendPersonRequest` into one function, since `getPerson` isn't doing anything by itself.